---
title: "cptac2 breast"
author: "Lizabeth Katsnelson"
date: "7/29/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(dplyr)
library(plyr)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(biomaRt)
library(data.table)
```

```{r}
### get all cnv segment files into one dataframe

cnv_segment <- read.table("cptac2_downloads/CPTAC_breast_data/cnv_segments.br.v1.5.2019-01-27/01BR001.corrected_GC.cnv", sep="\t", header=T)
cnv_segment <- cnv_segment[-c(1:483),]
colnames(cnv_segment)

setwd("cptac2_downloads/CPTAC_breast_data/cnv_segments.br.v1.5.2019-01-27/") # set working dir to all files
file_list <- list.files() # create list of all files in dir

for (file in file_list){
  
  print(file)
  
  # if df exists, add new file
  if (exists("cnv_segment")){
    temp_dataset <-read.table(file, sep="\t", header=T) # temp df for next file
    cnv_segment <- rbind(cnv_segment, temp_dataset) # join new file to existing df 
    rm(temp_dataset) # remove the temp df
  }
}

cnv_segment
write.table(cnv_segment, file = "cnv_segment_total.txt", sep = "\t", col.names = T, row.names = F, quote = F)


#cnv_segment_unique <- unique(cnv_segment$Sample)
#cnv_segment_unique # 128 samples
```





```{r}
### set working dir
setwd("~/Documents/GoogleSummerofCode/cptac2_downloads/CPTAC_breast_data/")

### read in all files
clinical <- read.table("Human__CPTAC_BRCA_2018__CPTAC__Clinical__Clinical__06 26 2019__CPTAC, BI, BCM__Clinical__BI_Clinical.tsi", sep='\t', header=T)
clinical

clinical_1_year_followup <- read.csv("S039_CPTAC_Breast_Confirmatory_Study_1year_Followup.csv", sep=',', header=T)
clinical_1_year_followup

sample <- read.csv("S039_Breast_Cancer_Prospective_Collection_Specimens_r1.csv", sep=',', header=T)
sample

mutation <- read.csv("Human__CPTAC_BRCA_2018__BI__Mutation__ExomeSeq__05 8 2018__Wash U__Gene__GATK_Pipeline.cbt", sep='\t', header=T)
mutation

cnv_segment <- read.table("cnv_segment_total.txt", sep='\t', header=T)
cnv_segment

cnv_log2 <- read.table("Human__CPTAC_BRCA_2018__BI__SCNV__ExomeSeq__05 31 2019__BI__Gene__GISTIC2_logratio.cct", sep='\t', header=T)
cnv_log2

cnv_threshold <- read.table("all_thresholded.by_genes.txt", sep='\t', header=T)
cnv_threshold

rna <- read.table("Human__CPTAC_BRCA_2018__BI__RNAseq__Illumina HiSeq RNA__05 31 2019__BI__Gene__FPKM_Normalizedlog2.cct", sep='\t', header=T)
rna

protein <- read.table("Human__CPTAC_BRCA_2018__BI__Proteome__TMT__05 25 2019__BI__Gene__Tumor_TMT_LogRatio.cct", sep='\t', header=T)
protein

phospho <- read.table("Human__CPTAC_BRCA_2018__BI__Phosphoproteome__TMT__05 25 2019__BI__Phosphosite__Tumor_TMT_LogRatio.cct", sep='\t', header=T)
phospho

acetyl <- read.table("Human__CPTAC_BRCA_2018__BI__Acetylome__TMT__05 25 2019__BI__Acetylation Site__Tumor_TMT_LogRatio.cct", sep='\t', header=T)
acetyl
```

---


### sample ID's
```{r}

clinical_samples <- data.frame(ID=colnames(clinical[2:123]), CLINICAL_PATIENT=rep(1))
clinical_samples$ID <- gsub(x = clinical_samples$ID, pattern = "X", replacement = "") # remove "X" from colnames string
clinical_samples$ID <- sort(clinical_samples$ID, decreasing=T)
clinical_samples

clinical_1_year_followup_samples <- data.frame(ID=clinical_1_year_followup$Participant.ID, CLINICAL_FOLLOWUP=rep(1))
clinical_1_year_followup_samples$ID <- sort(clinical_1_year_followup_samples$ID, decreasing=T)
clinical_1_year_followup_samples

sample_samples <- data.frame(ID=unique(sample$Participant_Identifier), SAMPLE=rep(1))
sample_samples$ID <- sort(sample_samples$ID, decreasing=T)
sample_samples

mutation_samples <- data.frame(ID=colnames(mutation[2:123]), MUTATION=rep(1))
mutation_samples$ID <- gsub(x = mutation_samples$ID, pattern = "X", replacement = "")
mutation_samples

cnv_segment_samples <- data.frame(ID=unique(cnv_segment$Sample), CNV_SEGMENT=rep(1))
cnv_segment_samples$ID <- sort(cnv_segment_samples$ID, decreasing=T)
cnv_segment_samples

cnv_log2_samples <- data.frame(ID=colnames(cnv_log2[2:123]), CNV_LOG2=rep(1))
cnv_log2_samples$ID <- gsub(x = cnv_log2_samples$ID, pattern = "X", replacement = "")
cnv_log2_samples$ID <- sort(cnv_log2_samples$ID, decreasing=T)
cnv_log2_samples

cnv_threshold_samples <- data.frame(ID=colnames(cnv_threshold[4:125]), CNV_THRESHOLD=rep(1))
cnv_threshold_samples$ID <- gsub(x = cnv_threshold_samples$ID, pattern = "X", replacement = "")
cnv_threshold_samples$ID <- gsub(x = cnv_threshold_samples$ID, pattern = "\\_.*", replacement = "") # remove everything after _
cnv_threshold_samples$ID <- sort(cnv_threshold_samples$ID, decreasing=T)
cnv_threshold_samples

rna_samples <- data.frame(ID=colnames(rna[2:123]), RNA=rep(1))
rna_samples$ID <- gsub(x = rna_samples$ID, pattern = "X", replacement = "")
rna_samples$ID <- sort(rna_samples$ID, decreasing=T)
rna_samples

protein_samples <- data.frame(ID=colnames(protein[2:123]), PROTEIN=rep(1))
protein_samples$ID <- gsub(x = protein_samples$ID, pattern = "X", replacement = "")
protein_samples$ID <- sort(protein_samples$ID, decreasing=T)
protein_samples

phospho_samples <- data.frame(ID=colnames(phospho[2:123]), PHOSPHO=rep(1))
phospho_samples$ID <- gsub(x = phospho_samples$ID, pattern = "X", replacement = "")
phospho_samples$ID <- sort(phospho_samples$ID, decreasing=T)
phospho_samples

acetyl_samples <- data.frame(ID=colnames(acetyl[2:123]), ACETYL=rep(1))
acetyl_samples$ID <- gsub(x = acetyl_samples$ID, pattern = "X", replacement = "")
acetyl_samples$ID <- sort(acetyl_samples$ID, decreasing=T)
acetyl_samples


all_samples <- join(clinical_samples, clinical_1_year_followup_samples, type="full")
all_samples <- join(all_samples, sample_samples, type="full")
all_samples <- join(all_samples, mutation_samples, type="full")
all_samples <- join(all_samples, cnv_segment_samples, type="full")
all_samples <- join(all_samples, cnv_log2_samples, type="full")
all_samples <- join(all_samples, cnv_threshold_samples, type="full")
all_samples <- join(all_samples, rna_samples, type="full")
all_samples <- join(all_samples, protein_samples, type="full")
all_samples <- join(all_samples, phospho_samples, type="full")
all_samples <- join(all_samples, acetyl_samples, type="full")

all_samples
# samples missing from all metadata files: 11BR069, 05BR058, 05BR055, 05BR052, 05BR051, 05BR031, 03BR012, 01BR044
### these samples only have CNV segment data, will remove them to avoid errors in cBio backend

# remove samples from CNV_SEGMENT df w/o metadata
cnv_segment$Sample <- as.character(cnv_segment$Sample)
delete_samples <- c("11BR069","05BR058","05BR055","05BR052","05BR051","05BR031","03BR012","01BR044")
cnv_segment_filt <- cnv_segment[-c(grep(paste(delete_samples, collapse="|"), cnv_segment$Sample)), ] # finds all rows with these sample IDs, deletes rows
cnv_segment_filt
unique(cnv_segment_filt$Sample) # 120 samples
```



## CLINICAL PATIENT
```{r}
# read in clinical data frame from linked omics 
#### variables for dataframes changed to preserve original format in first variable
meta <- clinical
as_tibble(meta)

## Transform data into proper format
meta_transposed <- data.frame(t(meta)) # transpose data frame - patient id's = rows, attriubutes = columns
colnames(meta_transposed) <- as.character(unlist(meta_transposed[1,])) # first row = colnames
meta_transposed <- rownames_to_column(meta_transposed) # rownames = first column
meta_transposed <-  meta_transposed[-1, ] # remove first row (turned into colnames)
meta_transposed
colnames(meta_transposed) <- c("PATIENT_ID", "STAGE", "PAM50", "NMF.v2.1", "CLINICAL_CLASSIFICATION", "ER", "PR", "HER2_REFINED", "HER2_STATUS_SATPATHY",
                               "HER1_AMP", "HER2_CLASS_SATPATHY", "AGE", "RACE", "NECROSIS", "TUMOR_CELLULARITY", "TOTAL_CELLULARITY", "CIN",
                               "MUTATION_LOAD", "MUTATED_GENES", "STEMNESS_SCORE", "CIBERSOTR_ABSOLUTE_SCORE", "CELL_IMMUNE_SCORE", "CELL_STROMA_SCORE", 
                               "CELL_MICROENVIRONMENT_SCORE", "ESTIMATE_IMMUNE_SCORE", "ESTIMATE_STROMAL_SCORE", "ESTIMATE_TUMOR_PURITY")

#Chromosome.INstability.index.CIN.
meta_transposed$PATIENT_ID <- gsub(x = meta_transposed$PATIENT_ID, pattern = "X", replacement = "") # get rid of "X" in front of patient identifier number 
meta_transposed$AGE <- as.numeric(as.character(meta_transposed$AGE)) 
meta_transposed$AGE <- round(meta_transposed$AGE/12) # change age to years
meta_transposed <- meta_transposed %>% mutate_all(as.character) # coerce all to characters 
meta_transposed[is.na(meta_transposed)] <- "[Not Available]" # change NA's to "[Not Available]"
as_tibble(meta_transposed)

################################################################################

# read in 1 year follow up clinical data frame from cptac
clinical_1_year <- clinical_1_year_followup
clinical_1_year <- clinical_1_year[, -c(2, 7:39, 41:45)] # remove unwanted columns
colnames(clinical_1_year) <- c("PATIENT_ID", "Days_Last_Contact", "OS_STATUS", 
                               "Days_to_Death", "TUMOR_STATUS", "Days_to_New_Tumor") # change colnames
clinical_1_year$OS_STATUS <- ifelse(clinical_1_year$OS_STATUS == "Living", "LIVING", "DECEASED")
clinical_1_year$OS_MONTHS <- ifelse(is.na(clinical_1_year$Days_Last_Contact), 
                                             clinical_1_year$Days_to_Death, 
                                             clinical_1_year$Days_Last_Contact) # combine days to death and days to last contact into one column
clinical_1_year$OS_MONTHS <- round(clinical_1_year$OS_MONTHS/12) # make days "months"
#count(clinical_1_year$TUMOR_STATUS) # get factor levels for tumor status (tumor free, with tumor, not reportd/unknown, and unknown tumor status)
clinical_1_year$DFS_STATUS <- ifelse(clinical_1_year$TUMOR_STATUS == "Tumor free", 
                                     "DiseaseFree", "Recurred/Progressed") # add proper factors for "DFS"

clinical_1_year$DFS_STATUS <- ifelse(clinical_1_year$TUMOR_STATUS == "Not Reported/Unknown" | clinical_1_year$TUMOR_STATUS == "Unknown tumor status", 
                                     "[Not Available]", 
                                     clinical_1_year$DFS_STATUS) # add Not Available for unknown tumor status

clinical_1_year$DFS_MONTHS <- ifelse(clinical_1_year$DFS_STATUS == "Recurred/Progressed", 
                                     clinical_1_year$Days_to_New_Tumor, 
                                     clinical_1_year$Days_Last_Contact) # add column for DFS_MONTHS

clinical_1_year$DFS_MONTHS <- round(clinical_1_year$DFS_MONTHS/12)
clinical_1_year$DFS_MONTHS <- ifelse(clinical_1_year$DFS_STATUS == "[Not Available]" | is.na(clinical_1_year$DFS_MONTHS), 
                                     "[Not Available]", 
                                     clinical_1_year$DFS_MONTHS) # add not avail
as_tibble(clinical_1_year)

################################################################################

breast_clinical <- join(clinical_1_year, meta_transposed, type="full")
breast_clinical <- breast_clinical %>% mutate_all(as.character) # coerce all to characters 
breast_clinical[is.na(breast_clinical)] <- "[Not Available]" # change NA's to "[Not Available]"
as_tibble(breast_clinical)

### format clinical patient text file
breast_clinical_patient <- data.frame(PATIENT_ID=breast_clinical$PATIENT_ID,
                                     AGE=breast_clinical$AGE,
                                     RACE=breast_clinical$RACE,
                                     OS_STATUS=breast_clinical$OS_STATUS,
                                     OS_MONTHS=breast_clinical$OS_MONTHS,
                                     DFS_STATUS=breast_clinical$DFS_STATUS,
                                     DFS_MONTHS=breast_clinical$DFS_MONTHS,
                                     PAM50=breast_clinical$PAM50,
                                     STAGE=breast_clinical$STAGE,
                                     CLINICAL_CLASSIFICATION=breast_clinical$CLINICAL_CLASSIFICATION,
                                     ER=breast_clinical$ER, 
                                     PR=breast_clinical$PR,
                                     HER2_REFINED=breast_clinical$HER2_REFINED,
                                     HER2_STATUS_SATPATHY=breast_clinical$HER2_STATUS_SATPATHY,
                                     HER1_AMP=breast_clinical$HER1_AMP,
                                     HER2_CLASS_SATPATHY=breast_clinical$HER2_CLASS_SATPATHY,
                                     MUTATED_GENES=breast_clinical$MUTATED_GENES)

breast_clinical_patient <- breast_clinical_patient[order(breast_clinical_patient$PATIENT_ID),] 
breast_clinical_patient$RACE <- gsub(x = breast_clinical_patient$RACE, pattern = "black.or.african.american", replacement = "BLACK OR AFRICAN AMERICAN")
breast_clinical_patient$RACE <- gsub(x = breast_clinical_patient$RACE, pattern = "asian", replacement = "ASIAN")
breast_clinical_patient$RACE <- gsub(x = breast_clinical_patient$RACE, pattern = "hispanic.or.latino", replacement = "HISPANIC OR LATINO")
breast_clinical_patient$RACE <- gsub(x = breast_clinical_patient$RACE, pattern = "white", replacement = "WHITE")

as_tibble(breast_clinical_patient)

# save df
#write.table(breast_clinical_patient, file = "cptac_breast_cbio_format/data_clinical_patient.txt", sep = "\t", col.names = T, row.names = F, quote = F)
#read.table("cptac_breast_cbio_format/data_clinical_patient.txt", sep='\t', header=T)

### header
#PATIENT_ID	AGE	RACE	OS_STATUS	OS_MONTHS	DFS_STATUS	DFS_MONTHS	PAM50	STAGE	CLINICAL_CLASSIFICATION	ER	PR	HER2_REFINED	HER2_STATUS_SATPATHY	HER1_AMP	HER2_CLASS_SATPATHY	MUTATED_GENES
#Patient unique identifier.	Patient age.	Patient race.	Overal patient survival status.	Overal patient survival months.	Overal disease free status.	Overal disease free months.	PAM50 status.	Tumor stage.	Clinical classification.	ER status.	PR status.	HER2 refined status.	HER2 status satpathy.	HER1 amplified.	HER2 class satpathy.	Number mutated genes. 
#STRING	NUMBER	STRING	STRING	NUMBER	STRING	NUMBER	STRING	STRING	STRING	NUMBER	NUMBER	NUMBER	NUMBER	NUMBER	STRING	NUMBER
#1	1	1	1	1	1	1	1	1	1	1	1	1	1	1	1	1
```



---

## CLNICAL SAMPLE 
```{r}
####sample_reformat[is.na(sample_reformat)] <- "[Not Available]" # change NA's to "[Not Available]"


#### read in CPTAC data portal meta data
sample_reformat <- sample 
sample_reformat <- sample_reformat[-c(grep("Adjacent_Normal", sample_reformat$Sample_Type)), ] # remove normal tissues
sample_reformat <- sample_reformat[, -c(2:3, 5:6)]
sample_reformat <- unique(sample_reformat)
sample_reformat <- data.frame(PATIENT_ID=sample_reformat$Participant_Identifier,
                              SAMPLE_ID=sample_reformat$Participant_Identifier,
                              CANCER_TYPE=rep("Breast Cancer"),
                              CANCER_TYPE_DETAILED=rep("Breast Invasive Carcinoma"), 
                              SAMPLE_TYPE=sample_reformat$Sample_Type)
sample_reformat


################################################

### sequenced samples
SEQUENCED  <- data.frame(PATIENT_ID = mutation_samples$ID, SEQUENCED = rep(1))
SEQUENCED # 122 samples

### samples with CNV data
COPY_NUMBER <- data.frame(PATIENT_ID = cnv_log2_samples$ID, COPY_NUMBER=rep(1))
COPY_NUMBER # 122 samples

### samples with mRNA data
MRNA_DATA <- data.frame(PATIENT_ID = rna_samples$ID, MRNA_DATA=rep(1))
MRNA_DATA # 122 samples

### samples with proteome data
PROTEIN <- data.frame(PATIENT_ID = protein_samples$ID, PROTEIN=rep(1))
PROTEIN # 122 samples

### samples with phosphoproteome data
PHOSPHOPROTEIN <- data.frame(PATIENT_ID = phospho_samples$ID, PHOSPHOPROTEIN=rep(1))
PHOSPHOPROTEIN # 122 samples

### samples with acetylome data
ACETYLOME <- data.frame(PATIENT_ID = acetyl_samples$ID, ACETYLOME=rep(1))
ACETYLOME # 122 samples

### all datatypes
all_data_types <- join(SEQUENCED, COPY_NUMBER, type="full")
all_data_types <- join(all_data_types, MRNA_DATA, type="full")
all_data_types <- join(all_data_types, PROTEIN, type="full")
all_data_types <- join(all_data_types, PHOSPHOPROTEIN, type="full")
all_data_types <- join(all_data_types, ACETYLOME, type="full")
all_data_types


### add sample info
total_meta <- join(sample_reformat, all_data_types, type="full")
total_meta <- total_meta %>% mutate_all(as.character) # coerce all to characters 
total_meta[124, 2] <- "CPT000814"
total_meta[125, 2] <- "CPT001846"
total_meta[6:11][is.na(total_meta[6:11])] <- 0
total_meta[1:5][is.na(total_meta[1:5])] <- "[Not Available]" # change NA's to "[Not Available]"
total_meta

#write.table(total_meta, file = "cptac_breast_cbio_format/data_clinical_sample.txt", sep = "\t", col.names = T, row.names = F, quote = F)

#### header
#PATIENT_ID	SAMPLE_ID	CANCER_TYPE	CANCER_TYPE_DETAILED	SAMPLE_TYPE	SEQUENCED	COPY_NUMBER	MRNA_DATA	PROTEIN	PHOSPHOPROTEIN	ACETYLOME
#Patient unique identifier.	Sample unique identifier.	Cancer type.	Detailed cancer type. 	Sample type.	Samples sequenced.	Samples with copy number data.	Samples with mRNA data.	Samples with protein data.	Samples with phosphoprotein data.	Samples with acetylome data.
#STRING	STRING	STRING	STRING	STRING	NUMBER	NUMBER	NUMBER	NUMBER	NUMBER	NUMBER
#1	1	1	1	1	1	1	1	1	1	1


SEQUENCED$PATIENT_ID
```


---

## CNV
```{r}
## CNV Segment scores
cnv_segment_reformat <- cnv_segment_filt
cnv_segment_reformat <- cnv_segment_reformat[, -7]
colnames(cnv_segment_reformat) <- c("ID", "chrom", "loc.start", "loc.end", "num.mark", "seg.mean")
cnv_segment_reformat$chrom <- gsub(x = cnv_segment_reformat$chrom, pattern = "chr", replacement = "") # remove "chr" from chromosome string
cnv_segment_reformat

#plyr::count(cnv_segment_reformat$chrom)
#write.table(cnv_segment_reformat, file = "cptac_breast_cbio_format/data_cna_hg19.seg", sep = "\t", col.names = T, row.names = F, quote = F)

################################################################################

## CNV GISTIC log2 ratio
cnv_log2_reformat <- cnv_log2
colnames(cnv_log2_reformat) <- gsub(x = colnames(cnv_log2_reformat), pattern = "X", replacement = "") # remove "X" from colnames string
colnames(cnv_log2_reformat)[1] <- "Hugo_Symbol"
cnv_log2_reformat$Hugo_Symbol <- as.character(cnv_log2_reformat$Hugo_Symbol)
cnv_log2_reformat

id_key <- read.table("Homo_sapiens.gene_info.txt", sep='\t', header=T) # get genes names to entrez id conversion
id_key

cnv_log2_reformat_ids <- join(id_key, cnv_log2_reformat, type="right")
cnv_log2_reformat_ids

colSums(is.na(cnv_log2_reformat_ids[1:2])) # 2500 NA's in entrez ID column

#write.table(cnv_log2_reformat_ids, file = "cptac_breast_cbio_format/data_log2_CNA.txt", sep = "\t", col.names = T, row.names = F, quote = F)

################################################################################

## CNV GISTIC Threhold Scores
cnv_threshold_reformat <- cnv_threshold
cnv_threshold_reformat <- cnv_threshold_reformat[, -c(2:3)]
colnames(cnv_threshold_reformat) <- gsub(x = colnames(cnv_threshold_reformat), pattern = "X", replacement = "")
colnames(cnv_threshold_reformat) <- gsub(x = colnames(cnv_threshold_reformat), pattern = "\\_.*", replacement = "")
colnames(cnv_threshold_reformat)[1] <- "Hugo_Symbol"
cnv_threshold_reformat 

cnv_threshold_reformat_ids <- join(id_key, cnv_threshold_reformat, type="right")
cnv_threshold_reformat_ids

colSums(is.na(cnv_threshold_reformat_ids[1:2])) # 2500 NA's in entrez ID column

write.table(cnv_threshold_reformat_ids, file = "cptac_breast_cbio_format/data_CNA.txt", sep = "\t", col.names = T, row.names = F, quote = F)

```

---

## RNA
```{r}
## mRNA
rna_reformat <- rna
colnames(rna_reformat) <- gsub(x = colnames(rna_reformat), pattern = "X", replacement = "") # remove "X" from colnames string
colnames(rna_reformat)[1] <- "Hugo_Symbol"
rna_reformat$Hugo_Symbol <- as.character(rna_reformat$Hugo_Symbol)
rna_reformat

rna_reformat_ids <- join(id_key, rna_reformat, type="right")
rna_reformat_ids

colSums(is.na(rna_reformat_ids[1:2])) # 1746 NA's in entrez ID column

#write.table(rna_reformat_ids, file = "cptac_breast_cbio_format/data_RNA_Seq_v2_expression_FPKM_log2.txt", sep = "\t", col.names = T, row.names = F, quote = F)

```

---

## Protein
```{r}
## PROTEOME
protein_reformat <- protein
colnames(protein_reformat) <- gsub(x = colnames(protein_reformat), pattern = "X", replacement = "") # remove "X" from colnames string
colnames(protein_reformat)[1] <- "Composite.Element.REF"
protein_reformat$Composite.Element.REF <- as.character(protein_reformat$Composite.Element.REF)
protein_reformat

### fix naming in protein name column 
prot <- data.frame(prot_string = protein_reformat$Composite.Element.REF, prot = protein_reformat$Composite.Element.REF) # new df with protein names
prot$prot_string <- paste(prot$prot_string, "|") # add pipe to protein name string
prot$total_name <- paste(prot$prot_string, prot$prot)
prot

### add proper names to protein dataset
protein_corrected <- cbind(prot$total_name, protein_reformat) # added new column - proper naming
protein_corrected <- protein_corrected[, -2] # remove previous names
colnames(protein_corrected)[1] <- "Composite.Element.REF" 
protein_corrected

################################################################################

## PHOSPHOPROTEOME
phospho_reformat <- phospho
colnames(phospho_reformat) <- gsub(x = colnames(phospho_reformat), pattern = "X", replacement = "") # remove "X" from colnames string
colnames(phospho_reformat)[1] <- "Composite.Element.REF"
phospho_reformat$Composite.Element.REF <- as.character(phospho_reformat$Composite.Element.REF)
phospho_reformat$Composite.Element.REF <- gsub(x = phospho_reformat$Composite.Element.REF, pattern = "x", replacement = "\\_p", ignore.case=F) #change x to _p for phospho
phospho_reformat$Composite.Element.REF <- gsub(x = phospho_reformat$Composite.Element.REF, pattern = "s", replacement = "", ignore.case=F) # get rid of s at end of string
phospho_reformat$Composite.Element.REF <- gsub(x = phospho_reformat$Composite.Element.REF, pattern = "t", replacement = "", ignore.case=F) # get rid of t at end of string
phospho_reformat$Composite.Element.REF <- gsub(x = phospho_reformat$Composite.Element.REF, pattern = "y", replacement = "", ignore.case=F) # get rid of y at end of string
phospho_reformat

### fix naming in phosphosite column 
protein_names <- data.frame(prot = phospho_reformat$Composite.Element.REF, phospho = phospho_reformat$Composite.Element.REF) # new df with prot names
protein_names$prot <- gsub(x = protein_names$prot, pattern = "\\_.*", replacement = "") # remove phosphosite to have just protein name
protein_names$prot <- paste(protein_names$prot, "|") # add pipe to protein name
protein_names$total_name <- paste(protein_names$prot, protein_names$phospho)
protein_names

### add proper names to phospho dataset
phospho_corrected <- cbind(protein_names$total_name, phospho_reformat) # added new column - proper naming
phospho_corrected <- phospho_corrected[, -2] # remove previous names
colnames(phospho_corrected)[1] <- "Composite.Element.REF" 
phospho_corrected

################################################################################

### Merge protein and phospho data to one dataframe
total_protein <- rbind(protein_corrected, phospho_corrected)
total_protein

#write.table(total_protein, file = "cptac_breast_cbio_format/data_protein_quantification.txt", sep = "\t", col.names = T, row.names = F, quote = F)


################################################################################

### add acetylome data
acetyl_reformat <- acetyl
colnames(acetyl_reformat) <- gsub(x = colnames(acetyl_reformat), pattern = "X", replacement = "") # remove "X" from colnames string
colnames(acetyl_reformat)[1] <- "Composite.Element.REF"
acetyl_reformat$Composite.Element.REF <- as.character(acetyl_reformat$Composite.Element.REF)
acetyl_reformat$Composite.Element.REF <- gsub(x = acetyl_reformat$Composite.Element.REF, pattern = "x", replacement = "\\_a", ignore.case=F) #change x to _a for acetyl
acetyl_reformat$Composite.Element.REF <- gsub(x = acetyl_reformat$Composite.Element.REF, pattern = "k", replacement = "", ignore.case=F) # get rid of k at end of string
acetyl_reformat

### fix naming in acetyle column 
protein_names <- data.frame(prot = acetyl_reformat$Composite.Element.REF, acetyl = acetyl_reformat$Composite.Element.REF) # new df with prot names
protein_names$prot <- gsub(x = protein_names$prot, pattern = "\\_.*", replacement = "") # remove acetyl site to have just protein name
protein_names$prot <- paste(protein_names$prot, "|") # add pipe to protein name
protein_names$total_name <- paste(protein_names$prot, protein_names$acetyl)
protein_names

### add proper names to phospho dataset
acetyl_corrected <- cbind(protein_names$total_name, acetyl_reformat) # added new column - proper naming
acetyl_corrected <- acetyl_corrected[, -2] # remove previous names
colnames(acetyl_corrected)[1] <- "Composite.Element.REF" 
acetyl_corrected

### Merge acetyl to all proetin dataframe
add_acetyl <- rbind(total_protein, acetyl_corrected)
add_acetyl

#write.table(add_acetyl, file = "cptac_breast_cbio_format/data_protein_quantification_WITH_ACETYLOME.txt", sep = "\t", col.names = T, row.names = F, quote = F)

```
